dbms/src/Functions/greatCircleArc.cpp

```diff
@@ -0,0 +1,74 @@
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+#include <ext/range.h>
+#include <cmath>
+
+namespace DB {
+
+class FunctionGreatCircleArc : public IFunction {
+public:
+	static constexpr auto name = "greatCircleArc";
+
+	static FunctionPtr create(const Context &) { return std::make_shared<FunctionGreatCircleArc>(); }
+
+	std::string getName() const override { return name; }
+
+	size_t getNumberOfArguments() const override { return 4; }
+	bool useDefaultImplementationForConstants() const override { return true; }
+
+	DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+	{
+		for (const auto i : ext::range(0, 4))
+		{
+			const auto arg = arguments[i].get();
+			if (!isNumber(arg))
+				throw Exception(
+						"Illegal type " + arg->getName() + " of argument " + std::to_string(i + 1) + " of function " + getName() + ". Must be a number",
+						ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+		}
+
+		return std::make_shared<DataTypeFloat64>();
+	}
+
+	void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) override
+	{
+		const auto col_lon1Deg = block.getByPosition(arguments[0]).column.get();
+		const auto col_lat1Deg = block.getByPosition(arguments[1]).column.get();
+		const auto col_lon2Deg = block.getByPosition(arguments[2]).column.get();
+		const auto col_lat2Deg = block.getByPosition(arguments[3]).column.get();
+
+		auto dst = ColumnVector<Float64>::create();
+		auto & dst_data = dst->getData();
+		dst_data.resize(input_rows_count);
+
+		for (const auto row : ext::range(0, input_rows_count))
+		{
+			const Float64 lon1 = rad_in_deg * col_lon1Deg->getUInt(row);
```

Comment:
And it's strange to get unsigned (1) integer (2) values of coordinates.

Reply:
Thank you, I've fixed it and rebased my commit history